www_src/pages/editor/editor.jsx

```diff
@@ -48,5 +48,96 @@ var Editor = React.createClass({
   }
 });
 
+var TextEditor = React.createClass({
+  mixins: [React.addons.LinkedStateMixin, Binding],
+  getInitialState: function () {
+    return {
+      fontSize: 18,
+      fontFamily: 'Roboto',
+      color: '#645839',
+      fontWeight: 'normal',
+      fontStyle: 'normal',
+      textDecoration: 'none',
+      textAlign: 'center'
+    };
+  },
+  onChange: function (id) {
+    return (e) => {
+      var state = {};
+      state[id] = e.target.checked;
+      this.setState(state);
+    }
+  },
+  render: function () {
+    var textStyleOptions = [
+      {
+        id: 'fontWeight',
+        icon: '../../img/B.svg',
+        falsey: 'normal',
+        truthy: 'bold'
+      },
+      {
+        id: 'fontStyle',
+        icon: '../../img/I.svg',
+        falsey: 'normal',
+        truthy: 'italic'
+      },
+      {
+        id: 'textDecoration',
+        icon: '../../img/U.svg',
+        falsey: 'none',
+        truthy: 'underline'
+      }
+    ];
+    var textAlignOptions = [
+      {
+        value: 'left',
+        icon: '../../img/align-left.svg',
+      },
+      {
+        value: 'center',
+        icon: '../../img/align-center.svg',
+      },
+      {
+        value: 'right',
+        icon: '../../img/align-right.svg',
+      }
+    ];
+    return (
+      <div id="editor">
+        <div className="editor-preview">
+          <p style={this.state}>Hello world</p>
+        </div>
+        <div className="editor-options">
+          <div className="form-group">
+            <label>Text Size</label>
+            <Range id="fontSize" min={10} max={32} unit="px" linkState={this.linkState} />
+          </div>
+          <div className="form-group">
+            <label>Font</label>
+            <select className="select" valueLink={this.linkState('fontFamily')}>
+              <option value="Roboto">Roboto</option>
+              <option value="Bitter">Bitter</option>
+              <option value="Pacifico">Pacifico</option>
+            </select>
+          </div>
+          <div className="form-group">
+            <label>Color</label>
+            <ColorGroup id="color" linkState={this.linkState} />
+          </div>
+          <div className="form-group">
+            <label>Text Style</label>
+            <CheckboxSet options={textStyleOptions} linkState={this.linkState} />
+          </div>
+          <div className="form-group">
+            <label>Text Alignment</label>
+            <Radio id="textAlign" options={textAlignOptions} linkState={this.linkState} />
+          </div>
+        </div>
+      </div>
+    );
+  }
+});
+
 // Render!
-render(Editor);
+render(window.location.hash === '#text' ? TextEditor : ImageEditor);
```

Comment:
temporary hack tbh.

there are a number of ways we can do this. probably the best way is to pass the params from the android router, and then render a child component based on what the type of the element is.

im planning on working on hooking this stuff up to real api responses etc next week.